### PR TITLE
add the jetty csrf filter to EmissaryServer

### PR DESCRIPTION
**Pull request by dev-mlb**

### CODE REVIEW

File: src/main/java/emissary/server/mvc/adapters/PlaceStarterAdapter.java [hunk 1]
```diff
@@ -71,12 +73,12 @@ public IServiceProviderPlace inboundCreatePlace() {
      */
     public EmissaryResponse outboundCreatePlace() {
 
+        final String placeUrl = getLocation().substring(0, getLocation().lastIndexOf("/") + 1);
         // Derive url from place location
-        final String url = getLocation().substring(0, getLocation().lastIndexOf("/") + 1) + CONTEXT + "/CreatePlace.action";
-
-        logger.debug("Creating post method for createPlace using URL " + url);
+        final String url = placeUrl + CONTEXT + "/CreatePlace.action";
```
Comment (fbruton) [inline src/main/java/emissary/server/mvc/adapters/PlaceStarterAdapter.java:78]: Unused variable.

**Review by fbruton — CHANGES_REQUESTED** [review]

**Review by fbruton — APPROVED** [review]

**Comment by jpdahlke** [comment]
just to confirm:  

Old endpoints:
/emissary/Pause.action -X GET
/emissary/Unpause.action -X GET
/emissary/Shutdown.action -X GET

New endpoints
/api/Pause -X POST
/api/Unpause -X POST
/api/Shutdown -X POST

File: src/main/java/emissary/command/ServiceCommand.java [hunk 2]
```diff
@@ -117,13 +125,23 @@ protected void stopService() {
...
+    }
+
+    /**
+     * Send a get request using the {@link EmissaryClient}
+     *
+     * @param endpoint the endpoint i.e. /api/health
      * @return the response object
      */
-    protected EmissaryResponse performAction(String actionEndpoint) {
+    protected EmissaryResponse performPost(String endpoint) {
```
Comment (jpdahlke) [inline src/main/java/emissary/command/ServiceCommand.java:141]: performPost, performGet and getEndpoint should probably all move to the HttpCommand

**Review by jpdahlke — CHANGES_REQUESTED** [review]
It appears that the stop command failed to properly shutdown a running server and look to be due to needing an update to StopCommand like:
`EmissaryResponse response = performPost("/api/Shutdown");`

**Comment by jpdahlke** [comment]
If you are in here, perhaps a log stating we are using csrf? It could be debug, but just something to show during startup.

File: src/main/java/emissary/command/ServiceCommand.java (hunk 2 above)
Comment (dev-mlb) [reply src/main/java/emissary/command/ServiceCommand.java]: good point...moved the methods to HttpCommand

**Comment by dev-mlb** [comment]
New endpoints updated to 
/api/pause -X POST
/api/unpause -X POST
/api/shutdown -X POST

**Comment by dev-mlb** [comment]
Stop command fixed...missed it since it's deprecated, so god catch. I debated removing the stop command, but I'll leave that for another pull request.

**Comment by dev-mlb** [comment]
Added debug statements for csrf on startup

**Review by jpdahlke — APPROVED** [review]

**Pull request merged**